soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -381,4 +381,13 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 	z_xt_ints_on(core_desc[cpu].intenable);
 }
 
+#ifdef CONFIG_ARCH_CPU_IDLE_CUSTOM
+
+void arch_cpu_idle(void)
+{
+	sys_trace_idle();
+	__asm__ volatile("waiti 0");
```

Comment:
Just remove this patch?  Just add your custom idle code once, no need to add a noop copy first.

Reply:
Also, and I do see that the ARCH_CPU_IDLE_CUSTOM has already merged, but the whole feature looks silly to me.  The Zephyr kernel **already** allows for indirection of CPU idle entry to the platform layers, that's why it's called "arch_"!

If the arch wants to further delegate, that's fine too, but there's no reason the core Zephyr code should need to care.

In this case, it would have been much simpler to make the xtensa arch_cpu_idle() a `__weak` function, document that fact, and then just implement your own in the SOC code.